### PR TITLE
Refactor: replace deprecated timezone_offset parameter with timezone

### PR DESCRIPTION
**Pull request by PooyaRaki**
Removes deprecated timezone_offset, Adds `timezone` as a replacement.

## What it solves

Resolves
https://github.com/safe-global/safe-client-gateway/issues/1024

## How this PR fixes it
A new query parameter, timezone, has been introduced. This parameter accepts only valid timezone strings (e.g., Europe/Berlin) and supersedes the deprecated timezone_offset, which previously accepted a numeric timezone difference.

## How to test it

- Review the transaction history list.
- Make sure the transactions are listed with the correct time and date, and verify that they are grouped by day properly.
- Ensure that transactions occurring around midnight are not mistakenly assigned to the following day.
For more info please visit https://github.com/safe-global/safe-client-gateway/issues/1024

## Checklist
* [ ] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

**Comment by PooyaRaki** [comment]
Currently, there are some errors present due to the SDK version not being updated. Please ensure to update the SDK version accordingly.

**Comment by PooyaRaki** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by katspaugh — COMMENTED** [review]
You can now install `@safe-global/safe-gateway-typescript-sdk@3.22.4-beta.1` and commit that as well so that we can test this PR.

**Comment by PooyaRaki** [comment]
> You can now install `@safe-global/safe-gateway-typescript-sdk@3.22.4-beta.1` and commit that as well so that we can test this PR.

@katspaugh Done ✅

File: package.json [hunk 1]
```diff
@@ -158,5 +158,6 @@
     "budgetPercentIncreaseRed": 20,
     "minimumChangeThreshold": 0,
     "showDetails": true
-  }
+  },
+  "packageManager": "yarn@1.22.22+sha512.a6b2f7906b721bba3d67d4aff083df04dad64c399707841b7acf00f6b133b7ac24255f2652fa22ae3534329dc6180534e98d17432037ff6fd140556e2bb3137e"
```
Comment (katspaugh) [inline package.json:162]: 🤔

Comment (katspaugh) [inline package.json:162]:
```suggestion
  }
```
Not sure why it got added but we don't seem to have needed it so far.

**Review by compojoom — APPROVED** [review]
👍

**Comment by katspaugh** [comment]
Tested, the timezone parameter is added and the date labels are correct. ✅

**Pull request merged**